--- src/Terraria/Terraria/GameContent/Tile_Entities/TEItemFrame.cs
+++ src/tModLoader/Terraria/GameContent/Tile_Entities/TEItemFrame.cs
@@ -1,5 +_,6 @@
 using System.IO;
 using Terraria.DataStructures;
+using Terraria.ID;
 
 namespace Terraria.GameContent.Tile_Entities
 {
@@ -71,8 +_,8 @@
 		}
 
 		public override void WriteExtraData(BinaryWriter writer, bool networkSend) {
-			writer.Write((short)item.netID);
-			writer.Write(item.prefix);
+			writer.Write((short)(item.netID >= ItemID.Count ? 0 : item.netID));
+			ModLoader.IO.ItemIO.WriteVanillaPrefix(item, writer);
 			writer.Write((short)item.stack);
 		}
 
